tensorrt_llm/_torch/bolt_profiling.py

```
@@ -0,0 +1,370 @@
+    Calling a mis-computed absolute address would SIGSEGV the workload (a long,
+    expensive run), so gate the raw call on dladdr agreeing the address belongs
+    to the expected library.
+    """
+    try:
+        libc = ctypes.CDLL(None)
+        info = _DlInfo()
+        if libc.dladdr(ctypes.c_void_p(addr), ctypes.byref(info)) == 0:
+            return False
+        fname = (info.dli_fname or b"").decode(errors="replace")
+        return os.path.basename(os.path.realpath(fname)) == os.path.basename(os.path.realpath(path))
+    except Exception:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)